aws/resource_aws_sns_application_test.go

```diff
@@ -0,0 +1,204 @@
+package aws
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/sns"
+)
+
+/**
+ Before running this test, a few ENV variables must be set:
+ GCM_API_KEY - Google Cloud Messaging API Key
+ APNS_SANDBOX_CREDENTIAL - Apple Push Notification Sandbox Private Key
+ APNS_SANDBOX_PRINCIPAL - Apple Push Notification Sandbox Certificate
+**/
+
+func TestAccAWSSNSApplication_gcm_create_update(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			if os.Getenv("GCM_API_KEY") == "" {
+				t.Fatal("GCM_API_KEY must be set.")
+			}
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSNSApplicationDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSSNSApplicationGCMConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"aws_sns_application.gcm_test", "name", "aws_sns_application_test"),
+					resource.TestCheckResourceAttr(
+						"aws_sns_application.gcm_test", "platform", "GCM"),
+					resource.TestCheckResourceAttr(
+						"aws_sns_application.gcm_test", "success_sample_rate", "100"),
+					resource.TestCheckResourceAttr(
+						"aws_sns_application.gcm_test", "created_topic", "arn:aws:sns:us-east-1:638386993804:endpoint-created-topic"),
```

Comment:
The `arn` values here will change based on which user runs the acceptance test. Can use `TestCheckResourceAttrSet()` to verify that the attribute was set to a value. For explicit attribute checking, I believe we have a helper function in the `aws` provider to build an `arn` regardless of user-id or govcloud, etc.